pkg/telemetry/phonehome/segment/segment.go

```diff
@@ -130,6 +131,17 @@ func makeDeviceContext(o *telemeter.CallOptions) *segment.Context {
 			Type: o.ClientType,
 		}
 	}
+
+	if o.UserID == "" {
+		// Add "Server" suffix to the platform of the backend initiated events:
+		if ctx == nil {
+			ctx = &segment.Context{}
+		}
+		if ctx.Device.Type == "" {
+			ctx.Device.Type = t.clientType
+		}
+		ctx.Device.Type += " Server"
```

Comment:
Doesn't this add ` Server` multiple times when `makeDeviceContext` is called consecutively?

Reply:
No, because this method returns a new instance of context, it doesn't modify an existing one.